src/ddbToEs/ddbToEsHelper.ts

```diff
@@ -42,57 +43,89 @@ export default class DdbToEsHelper {
         });
     }
 
-    async createIndexAndAliasIfNotExist(indexName: string) {
-        logger.debug('entering create index function');
+    async createIndexAndAliasIfNotExist(aliases: Set<string>) {
+        if (aliases.size === 0) {
+            return;
+        }
+
+        const listOfAliases = Array.from(aliases);
+        const { body: allFound } = await this.ElasticSearch.indices.existsAlias({
+            name: listOfAliases,
+            expand_wildcards: 'all',
+        });
+        if (allFound) {
+            // All needed aliases exist
+            return;
+        }
+
+        logger.debug('There are missing aliases');
+
+        const indicesToCreate: Set<string> = new Set();
+        listOfAliases.forEach((alias: string) => {
+            indicesToCreate.add(alias.substring(0, alias.length - 6)); // remove '-alias'
+        });
+        const aliasesToCreate: Set<string> = new Set(aliases);
+
+        const { body: indices } = await this.ElasticSearch.indices.getAlias();
+        // for each index and alias found remove from set
+        Object.entries(indices).forEach(([k, v]) => {
+            indicesToCreate.delete(k);
+            Object.keys((v as any).aliases).forEach((aliasNames: string) => {
+                aliasesToCreate.delete(aliasNames);
+            });
+        });
         try {
-            const indexExistResponse = await this.ElasticSearch.indices.exists({ index: indexName });
-            logger.debug(indexExistResponse);
-            if (!indexExistResponse.body) {
-                // Create Index
-                const params = {
-                    index: indexName,
-                    body: {
-                        mappings: {
-                            properties: {
-                                id: {
-                                    type: 'keyword',
-                                    index: true,
-                                },
-                                resourceType: {
-                                    type: 'keyword',
-                                    index: true,
-                                },
-                                _references: {
-                                    type: 'keyword',
-                                    index: true,
-                                },
-                                documentStatus: {
-                                    type: 'keyword',
-                                    index: true,
+            const promises: any[] = [];
+            Array.from(indicesToCreate).forEach((index: string) => {
+                const alias = `${index}-alias`;
+                // Only create index when we also need to create an alias
+                if (aliasesToCreate.has(alias)) {
```

Comment:
The scenario I am trying to avoid is if/when index corruption happens and a new index is name something non-traditional `patient-new` in this situation we would have `patient` in our index set but `patient-alias` isn't so we shouldn't create `patient` index. 

The assumption I am making here is FWoA really only cares if the Alias is set up can connected to *a nonspecific* index

Reply:
I see. I wasn't considering index creation outside of this lambda. That's fair.